Source/Core/GoogleEarthEnterpriseTerrainData.js

```diff
@@ -0,0 +1,521 @@
+        var childTileMask = googleChildTileMask & 3; // Bottom row is identical
+        childTileMask |= (googleChildTileMask & 4) ? 8 : 0; // NE
+        childTileMask |= (googleChildTileMask & 8) ? 4 : 0; // NW
+
+        this._childTileMask = childTileMask;
+
+        this._createdByUpsampling = defaultValue(options.createdByUpsampling, false);
+
+        this._skirtHeight = undefined;
+        this._bufferType = this._buffer.constructor;
+        this._mesh = undefined;
+        this._minimumHeight = undefined;
+        this._maximumHeight = undefined;
+        this._vertexCountWithoutSkirts = undefined;
+        this._skirtIndex = undefined;
+    }
+
+    defineProperties(GoogleEarthEnterpriseTerrainData.prototype, {
+        /**
+         * The water mask included in this terrain data, if any.  A water mask is a rectangular
+         * Uint8Array or image where a value of 255 indicates water and a value of 0 indicates land.
+         * Values in between 0 and 255 are allowed as well to smoothly blend between land and water.
+         * @memberof GoogleEarthEnterpriseTerrainData.prototype
+         * @type {Uint8Array|Image|Canvas}
+         */
+        waterMask : {
+            get : function() {
+                return undefined;
+            }
+        }
+    });
+
+    var taskProcessor = new TaskProcessor('createVerticesFromGoogleEarthEnterpriseBuffer');
+
+    /**
+     * Creates a {@link TerrainMesh} from this terrain data.
+     *
+     * @private
+     *
+     * @param {TilingScheme} tilingScheme The tiling scheme to which this tile belongs.
+     * @param {Number} x The X coordinate of the tile for which to create the terrain data.
+     * @param {Number} y The Y coordinate of the tile for which to create the terrain data.
+     * @param {Number} level The level of the tile for which to create the terrain data.
+     * @param {Number} [exaggeration=1.0] The scale used to exaggerate the terrain.
+     * @returns {Promise.<TerrainMesh>|undefined} A promise for the terrain mesh, or undefined if too many
+     *          asynchronous mesh creations are already in progress and the operation should
+     *          be retried later.
+     */
+    GoogleEarthEnterpriseTerrainData.prototype.createMesh = function(tilingScheme, x, y, level, exaggeration) {
+        //>>includeStart('debug', pragmas.debug);
+        if (!defined(tilingScheme)) {
+            throw new DeveloperError('tilingScheme is required.');
+        }
+        if (!defined(x)) {
+            throw new DeveloperError('x is required.');
+        }
+        if (!defined(y)) {
+            throw new DeveloperError('y is required.');
+        }
+        if (!defined(level)) {
+            throw new DeveloperError('level is required.');
+        }
+        //>>includeEnd('debug');
+
+        var ellipsoid = tilingScheme.ellipsoid;
+        var nativeRectangle = tilingScheme.tileXYToNativeRectangle(x, y, level);
```

Comment:
Should be be using scratch values and result parameters here (also in cartographicToCartesian below).